include/can/core/messages.hpp

```diff
@@ -45,6 +45,44 @@ struct HeartbeatResponse {
     }
 };
 
+struct DeviceInfoRequest {
+    static const auto id = MessageId::device_info_request;
+
+    template <bit_utils::ByteIterator Input, typename Limit>
+    static auto parse(Input body, Limit limit) -> DeviceInfoRequest {
+        return DeviceInfoRequest{};
+    }
+
+    template <bit_utils::ByteIterator Output, typename Limit>
+    auto serialize(Output body, Limit limit) const -> uint8_t {
+        return 0;
+    }
+};
+
+struct DeviceInfoResponse {
+    static const auto id = MessageId::device_info_response;
+
+    NodeId node_id;
+    uint32_t version;
```

Comment:
Preliminary values for the response. Enough to id the sub system and its version.

is uint32 good enough for version?

Reply:
IMO we should set this up for a couple more things than just version, and then can care about version in general a little less. I don't think it's necessarily critical to get right the first time, but we could do something like
- two bits of build type (dev, testing, release)
- a byte or two of like message schema version
- a four byte incrementing version seems fine i suppose
- a byte or two for well-known-id system identification through some centrally defined enum
- at some point we'll want a serial number probably
- a hardware revision unless we want to fold that into the well-known-id

Reply:
Good thoughts. I'll add this comment as todo and merge as is.